framework/v2/guides/providers.md

```diff
@@ -68,3 +68,93 @@ Error example:
   message: 'Sender does not have sufficient balance.'
 }
 ```
+
+## Create your own provider
+
+We have some predefined providers so you can use out of the box but if you have some other way you want to connect to blockchain (e.g. you want to do raw transaction) you can write you own provider. This will show you the basic principle of how to create you own provider.
+
+All providers have the same structure which is provided by `GenericProvider`. Generic provider has a `client` which is the main part that we need to override to achieve custom functionality.
```

Comment:
All providers have the same structure, provided by `GenericProvider`. GenericProvider has a `client` which is the main part that we need to override to achieve custom functionality.